snapshotter/demux/cache/snapshotter_cache_test.go

```diff
@@ -67,6 +82,54 @@ func successfulWalk(ctx context.Context, info snapshots.Info) error {
 	return nil
 }
 
+func removeFunctionOnEmptyCache(uut *SnapshotterCache) error {
+	if err := uut.RemoveAll(context.Background(), "SnapshotKey"); err != nil {
+		return fmt.Errorf("RemoveAll on empty cache incorrectly resulted in error")
+	}
+	return nil
+}
+
+func removeFunctionToAllCachedSnapshotters(uut *SnapshotterCache) error {
+	if _, err := uut.Get(context.Background(), "Snapshotter-A", getSnapshotterOkFunction); err != nil {
+		return fmt.Errorf("Adding snapshotter A to empty cache incorrectly resulted in error [%w]", err)
+	}
+	if _, err := uut.Get(context.Background(), "Snapshotter-B", getSnapshotterOkFunction); err != nil {
+		return fmt.Errorf("Adding snapshotter B to cache incorrectly resulted in error [%w]", err)
+	}
+	if err := uut.RemoveAll(context.Background(), "SnapshotKey"); err != nil {
+		return fmt.Errorf("RemoveAll on populated cache incorrectly resulted in error")
```

Comment:
Should we be wrapping errors in these functions for debugging?

Reply:
Oof, yes good catch.